src/components/Header.jsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import './Header.scss';
+
+const Header = () => {
+  const [select, setSelect] = useState(1);
+  const [widthScreen, setWidthScreen] = useState(window.innerWidth);
+  useEffect(() => {
+    window.addEventListener('resize', () => setWidthScreen(window.innerWidth));
+  }, []);
+
+  const [menuDisplay, setMenuDisplay] = useState(window.innerWidth > 800);
+  useEffect(() => {
+    document.querySelector('.menuContainer').style.display = menuDisplay
+      ? 'flex'
+      : 'none';
+  }, [menuDisplay]);
+
+  const handleMenu = (click) => {
+    if (click === 'checkbox' || click === 'resize') {
+      setMenuDisplay(!menuDisplay);
+    } else if (click === 'link' && widthScreen <= 800) {
+      setMenuDisplay(false);
+    }
+    return null;
+  };
+
+  useEffect(() => {
+    if (widthScreen > 800 && menuDisplay === false) {
+      handleMenu('resize');
+    } else if (widthScreen <= 800 && menuDisplay === true) {
+      handleMenu('resize');
+    }
+  }, [widthScreen]);
+
+  return (
+    <header>
+      <nav>
+        <Link
+          onClick={() => {
+            setSelect(1);
+            handleMenu('link');
+          }}
+          className="navLogo"
+          to="/"
+        >
+          <img
+            src={`/img/${widthScreen > 800 ? 'logoDesktop' : 'logoPhone'}.svg`}
+            alt="logo"
+          />
+        </Link>
+        <label htmlFor="checkBurger" className="burger">
+          <input
+            type="checkbox"
+            id="checkBurger"
+            checked={menuDisplay}
+            onChange={() => handleMenu('checkbox')}
+          />
+          <span className="line1" />
+          <span className="line2" />
+          <span className="line3" />
+        </label>
+        <div className="menuContainer">
+          <ul className="navLinks">
+            <li>
+              <Link
+                onClick={() => {
+                  setSelect(1);
+                  handleMenu('link');
+                }}
+                className={`navLink ${select === 1 && 'selected'}`}
```

Comment:
Aww no no, there's a custom [NavLink](https://reactrouter.com/web/api/NavLink) component in react-router which does exactly that, you don't recode this whole behaviour !
(Consider I said the same on every navigation link in yoiur menu ;-) )